docs/DafnyRef/Modules.md

```diff
@@ -453,6 +453,25 @@ in `B` nor as `Z.a` in C would be valid, because `a` is not in `Z`.
 The default export set is important in the resolution of qualified
 names, as described in [Section 4.8](#sec-name-resolution).
 
+There are a few unusual cases to be noted:
+- an export set can be completely empty, as in `export Nothing`
+- an anonymous export set can be completely empty, as in `export`
+- an export set declaration followed by an extreme predicate declaration looks like this:
+`export least predicate() { true }`
+In this case the `least` (or `greatest`) is the identifier naming the export set.
+Consequently, `export least predicate[nat]() { true }` is illegal because `[nat]` cannot be part of a non-extreme predicate.
+Also, it is not possible to declare an anonymous, empty export set immediately prior to a declaration of an extreme predicate,
+because the `least` or `greatest` is parsed as the export set identifier. The workaround for this situation is to reorder the declarations.
```

Comment:
I just thought of a better or more natural workaround: Don't omit the name of the eponymous export set, but instead give its name explicitly. That is, change `export` to `export M` if this declaration is placed inside a module named `M`. (Alright, you still have to reorder the declarations inside the module if your module happens to be called `least`.)

Reply:
Revised accordingly.